internal/ethapi/api.go

```diff
@@ -1432,7 +1442,7 @@ func (s *PublicTransactionPoolAPI) SendRawPrivateTransaction(ctx context.Context
 	}
 
 	txHash := []byte(tx.Data())
-	isPrivate := args.PrivateFor != nil
+	isPrivate := (args.PrivateFor != nil) && tx.IsPrivate()
```

Comment:
In addition to checks for PrivateFor args is the check for `tx.IsPrivate()` required

Reply:
The reason for the additional `tx.IsPrivate()` check, is for the case where a transaction is submitted with `PrivateFor` set, but where the transaction is not signed with the correct private signer. In this case we should reject the tx right away.